plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java

```diff
@@ -35,121 +35,221 @@
+                Script.runSimpleBashScript("umount " + pool._mountDestPath);
+                rbdstoragePool.remove(uuid);
             }
         }
     }
 
     public List<NfsStoragePool> getStoragePools() {
-        synchronized (storagePool) {
-            return new ArrayList<>(storagePool.values());
+        synchronized (nfsstoragePool) {
+            return new ArrayList<>(nfsstoragePool.values());
+        }
+    }
+
+    public List<RbdStoragePool> getRbdStoragePools() {
+        synchronized (rbdstoragePool) {
+            return new ArrayList<>(rbdstoragePool.values());
         }
     }
 
     public NfsStoragePool getStoragePool(String uuid) {
-        synchronized (storagePool) {
-            return storagePool.get(uuid);
+        synchronized (nfsstoragePool) {
+            return nfsstoragePool.get(uuid);
+        }
+    }
+
+    public RbdStoragePool getRbdStoragePool(String uuid) {
+        synchronized (rbdstoragePool) {
+            return rbdstoragePool.get(uuid);
         }
     }
 
     protected void runHeartBeat() {
-        synchronized (storagePool) {
-            Set<String> removedPools = new HashSet<>();
-            for (String uuid : storagePool.keySet()) {
-                NfsStoragePool primaryStoragePool = storagePool.get(uuid);
-                StoragePool storage;
-                try {
-                    Connect conn = LibvirtConnection.getConnection();
-                    storage = conn.storagePoolLookupByUUIDString(uuid);
-                    if (storage == null || storage.getInfo().state != StoragePoolState.VIR_STORAGE_POOL_RUNNING) {
-                        if (storage == null) {
-                            s_logger.debug(String.format("Libvirt storage pool [%s] not found, removing from HA list.", uuid));
-                        } else {
-                            s_logger.debug(String.format("Libvirt storage pool [%s] found, but not running, removing from HA list.", uuid));
+        if(nfsstoragePool != null && !nfsstoragePool.isEmpty()) {
+            synchronized (nfsstoragePool) {
+                Set<String> removedPools = new HashSet<>();
+                for (String uuid : nfsstoragePool.keySet()) {
+                    NfsStoragePool primaryStoragePool = nfsstoragePool.get(uuid);
+                    StoragePool storage;
+                    try {
+                        Connect conn = LibvirtConnection.getConnection();
+                        storage = conn.storagePoolLookupByUUIDString(uuid);
+                        if (storage == null || storage.getInfo().state != StoragePoolState.VIR_STORAGE_POOL_RUNNING) {
+                            if (storage == null) {
+                                s_logger.debug(String.format("Libvirt storage pool [%s] not found, removing from HA list.", uuid));
+                            } else {
+                                s_logger.debug(String.format("Libvirt storage pool [%s] found, but not running, removing from HA list.", uuid));
```

Comment:
```suggestion
                                s_logger.debug(String.format("Libvirt storage pool [%s] was found, but it is not running, removing it from HA list.", uuid));
```

Reply:
@GutoVeronezi   I modified the code as you requested.